src/Refinery/String/Levenshtein.php

```diff
@@ -0,0 +1,138 @@
+     */
+    public function __construct(string $str = "", int $maximumDistance = 0,
+                                float $costIns = 1.0, float $costRep = 1.0, float $costDel = 1.0)
+    {
+        $this->mainStr= $str;
+        $this->maximumDistance = $maximumDistance;
+        $this->costIns = $costIns;
+        $this->costRep = $costRep;
+        $this->costDel = $costDel;
+    }
+
+    /**
+     * Levenshtein function alternative code as mentioned in the bug report:
+     * https://mantis.ilias.de/view.php?id=17861
+     * Original code under MIT-License:
+     * https://github.com/GordonLesti/levenshtein/blob/master/src/Levenshtein.php
+     *
+     * Runtime improvements have been added:
+     * A interruption that triggers when the difference in length is bigger than the allowed maximum distance
+     * A maximum distance that interrupts the execution of the algorithm, if it already is worse than the allowed
+     *
+     * @param string $str string which is used for a repeated distance calculation
+     * @return float with Levenshtein distance, if an interrupt happens earlier than the return value is a -1
+     */
+    protected function levenshtein(string $str): float
+    {
+        $matrix = [];
+        $str1Array = $this->StringToArray($this->mainStr);
+        $str2Array = $this->StringToArray($str);
+        $str1Length = count($str1Array);
+        $str2Length = count($str2Array);
+
+        // if the difference between string length is bigger than the maximum allowed levenshtein distance
+        // the code can be skipped
+        if (abs($str1Length - $str2Length) > $this->maximumDistance && $this->maximumDistance != 0){
+            return -1.0;
+        }
+
+        $row = [];
+        $row[0] = 0.0;
+        for ($j = 1; $j < $str2Length + 1; $j++) {
+            $row[$j] = $j * $this->costIns;
+        }
+
+        $matrix[0] = $row;
+        for ($i = 0; $i < $str1Length; $i++) {
+            $row = [];
+            $row[0] = ($i + 1) * $this->costDel;
+            for ($j = 0; $j < $str2Length; $j++) {
+                $row[$j + 1] = min(
+                    $matrix[$i][$j + 1] + $this->costDel,
+                    $row[$j] + $this->costIns,
+                    $matrix[$i][$j] + ($str1Array[$i] === $str2Array[$j] ? 0.0 : $this->costRep)
+                );
+            }
+            // maximum distance reached
+            if (min($row) > $this->maximumDistance && $this->maximumDistance != 0){
+                return -1.0;
+            }
+            $matrix[$i + 1] = $row;
+        }
+        return $matrix[$str1Length][$str2Length];
+    }
+
+    /**
+     * Helper function for levenshtein distance calculation, used to convert strings into arrays.
+     *
+     * @param string $str the string that is converted into an array
+     * @return array an array containing the characters of the string, each in a single cell
+     */
+    private function StringToArray(string $str): array
```

Comment:
```suggestion
    private function stringToArray(string $str): array
```

Maybe `stringToChars` or `stringToCharCollection` would be a better name.